lib/fabrik/Fabrik.js

```diff
@@ -31,12 +34,17 @@ class Fabrik {
       .then(managerConstructor => managerConstructor.load(plan));
   }
 
-  static createInstance(instance_id, service_id, plan_id) {
+  static createInstance(instance_id, service_id, plan_id, context) {
     const plan = catalog.getPlan(plan_id);
     assert.strictEqual(service_id, plan.service.id);
     return this
       .createManager(plan)
-      .then(manager => manager.createInstance(instance_id));
+      .then(manager => {
+        const platformManager = !context ? undefined : (context.platform === CONST.CONSUMING_ENVIRONMENT.CF ?
```

Comment:
I would probably move this to BasePlatformManager and ask it to give an instance from its static method.

Reply:
Let's decide on resolving this https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/131#discussion_r167481063 first. Then we can decide about instantiation of platform manager.